### PR TITLE
Documentation: manually add IDs to all sections

### LINKED ISSUE #1315: Documentation: sections/chapters should have IDs in hand-written XML
The XML files for most of the docs are auto-generated, but the first few chapters are hand-written.

In the hand-written part, all chapters/sections should have explicit IDs.

Since they don't currently have IDs, a new random ID is generated each time the website is built. Thus links like this are not stable (they change on each rebuild):

https://igraph.org/c/doc/igraph-Tutorial.html#idm209453291280

Notice `#idm209453291280`

### PR DESCRIPTION
**Pull request by szhorvat**
This fixes #1315

### CODE REVIEW

**Comment by szhorvat** [comment]
Because of the force-pushes above, I'm just noting that this is ready to merge now. I verified that there are no more random section IDs, except in the license.

File: doc/adjlist.xxml [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 <!-- doxrox-include about_adjlists -->
 
-<section><title>Adjacent vertices</title>
+<section id="adjacenct-vertices"><title>Adjacent vertices</title>
```
Comment (ntamas) [inline doc/adjlist.xxml:12]: There is a typo here (`adjacenct` vs `adjacent`)

File: doc/flows.xxml [hunk 2]
```diff
@@ -24,24 +24,24 @@
...
-<section><title>Connectivity</title>
+<section id="connectivity"><title>Connectivity</title>
 <!-- doxrox-include igraph_st_edge_connectivity -->
 <!-- doxrox-include igraph_edge_connectivity -->
 <!-- doxrox-include igraph_st_vertex_connectivity -->
 <!-- doxrox-include igraph_vertex_connectivity -->
 </section>
 
-<section><title>Edge- and Vertex-Disjoint Paths</title>
+<section id="edge-and-vertexdisjoint-paths"><title>Edge- and Vertex-Disjoint Paths</title>
```
Comment (ntamas) [inline doc/flows.xxml:34]: Please add a dash between `vertex` and `disjoint`

**Review by ntamas — CHANGES_REQUESTED** [review]
I have spotted two typos (see comments), otherwise it's okay -- thanks a lot!

Feel free to merge when the typos are fixed.

**Comment by szhorvat** [comment]
I ran a spell check on some, but not all IDs, and made some more corrections. The dashes went missing because part of the job was automated and my script forgot about `-` characters. I also removed the `igraph-` prefixes in `error.xxml` and added in some dashes.

Pushed separate commit so you can see the changes. We can then squash-and-merge.

**Comment by ntamas** [comment]
Thanks!

**Pull request merged**